pkg/awsManager/vpcVpnManager.go

```diff
@@ -0,0 +1,542 @@
+package awsManager
+
+import (
+	"errors"
+	"fmt"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/elb"
+	"github.com/aws/aws-sdk-go/service/elbv2"
+	clientpkg "github.com/openshift/aws-account-shredder/pkg/aws"
+)
+
+func ListVPCforDeletion(client clientpkg.Client) ([]*string, error) {
+
+	var vpcToBeDeleted []*string
+	vpcList, err := client.DescribeVpcs(&ec2.DescribeVpcsInput{})
+	if err != nil {
+		fmt.Println("Error:", err)
+		fmt.Println("Error: Cant list VPC ")
+		return nil, errors.New("error")
+	}
+
+	for _, vpcs := range vpcList.Vpcs {
+		if *vpcs.IsDefault == false {
+			//fmt.Print(vpcs)											// for debugging
+			vpcToBeDeleted = append(vpcToBeDeleted, vpcs.VpcId)
+		}
+	}
+
+	return vpcToBeDeleted, nil
+}
+func CleanVpcInstances(client clientpkg.Client) {
+	vpcToBeDeleted, err := ListVPCforDeletion(client)
+	if err != nil {
+		fmt.Println("cant list VPC")
+		return
+	}
+	fmt.Println(vpcToBeDeleted)
+	_ = DeleteVpcInstances(client, vpcToBeDeleted)
+
+	// need to clear VPN connection now , VPC gateway has been detached already by now (function call in -> deleteVpcInstances()) , if not this will throw an error .
+	_ = DeleteVpnConnections(client)
+}
+func DeleteVpcInstances(client clientpkg.Client, vpcToBeDeleted []*string) error {
+
+	errFlag := false
+	for _, vpcId := range vpcToBeDeleted {
+
+		//need to clean out the dependencies
+		// EC2 + S3 are already cleaned out by now
+		// delete vpc endpoints
+		err := DeleteVpcEndpoint(client, vpcId)
```

Comment:
For all of these resources, do they actually need to be delete for the VPC to be deleted? Or just detached?

Reply:
Some resources ( like network interface ) cant be deleted until is is detached. Others , we can just delete them off.

Reply:
Right, my question is do we need to delete them?

Reply:
@fahlmant what's the reason to detach without delete?  Or would we come back and delete at a later time or something like that?

Reply:
@iamkirkbater Yeah, just seeing what the exact minimum number of steps it takes to actually delete the VPC/VPNs since that's the goal here. Not opposed to deleting the other resources, but this one function to clean is huge compared to s3, ec2, etc

Reply:
yeah , we need to delete them. ( for example the network interface ). based on my understanding , we need to detach and delete network interface. we need to do both. If we dont detach , it wont delete. If , it does not delete , VPC wont be deleted.